File: leetsync/2215-finding-3-digit-even-numbers/finding-3-digit-even-numbers.go
func findEvenNumbers(digits []int) []int {
    freqs := [10]int{};
    even := 0;

    for _, digit := range digits {
        freqs[digit]++;
        if digit % 2 == 0 {
            even++;
        }
    }

    if even == 0 {
        return []int{};
    }

    result := []int{}
    for x := 100; x <= 999; x += 2 {
        freq := freqs;
        x0 := x % 10;
        x1 := (x / 10) % 10;
        x2 := x / 100;

        freq[x0]--;
        if freq[x0] < 0 {
            continue;
        }

        freq[x1]--;
        if freq[x1] < 0 {
            continue;
        }

        freq[x2]--;
        if freq[x2] < 0 {
            continue;
        }

        result = append(result, x);
    }

    return result;
}

File: leetsync/2215-finding-3-digit-even-numbers/finding-3-digit-even-numbers.java
class Solution {
    public int[] findEvenNumbers(int[] digits) {
        List<Integer> result = new ArrayList<>();
        int[] freq = new int[10];
        for (int digit : digits) {
            freq[digit]++;
        }
        for (int i = 1; i <= 9; i++) {
            if (freq[i] == 0) {
                continue;
            }
            freq[i]--;
            for (int j = 0; j <= 9; j++) {
                if (freq[j] == 0) {
                    continue;
                }
                freq[j]--;
                for (int k = 0; k <= 8; k += 2) {
                    if (freq[k] == 0) {
                        continue;
                    }
                    freq[k]--;
                    int num = i * 100 + j * 10 + k;
                    result.add(num);
                    freq[k]++;
                }
                freq[j]++;
            }
            freq[i]++;
        }
        int[] resultArray = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            resultArray[i] = result.get(i);
        }
        return resultArray;
    }
}